raiden/storage/sqlite.py

```diff
@@ -1376,11 +1376,24 @@ def get_light_client_messages(self, from_message, light_client):
         cursor = self.conn.cursor()
         cursor.execute(
             """
-            SELECT identifier, message_order, unsigned_message, signed_message, light_client_payment_id, internal_msg_identifier, message_type
-            FROM light_client_protocol_message
-            WHERE internal_msg_identifier >= ?
-            AND light_client_address = ?
-            ORDER BY light_client_payment_id, message_order ASC
+            SELECT lcpm1.identifier,
+                   lcpm1.message_order,
+                   lcpm1.unsigned_message,
+                   lcpm1.signed_message,
+                   lcpm1.light_client_payment_id,
+                   lcpm1.internal_msg_identifier,
+                   lcpm1.message_type
+            FROM light_client_protocol_message lcpm1
+            WHERE lcpm1.internal_msg_identifier >= ?
+            AND lcpm1.light_client_address = ?
+            AND lcpm1.internal_msg_identifier NOT IN (
+                SELECT lcpm2.internal_msg_identifier
+                FROM light_client_protocol_message lcpm2
+                WHERE lcpm2.internal_msg_identifier >= ?
+                AND lcpm2.message_type IN ('SettlementRequired', 'RequestRegisterSecret', 'UnlockLightRequest')
+                AND lcpm2.signed_message IS NOT NULL
+            )
```

Comment:
```suggestion
             AND (NOT lcpm1.message_type IN ('SettlementRequired', 'RequestRegisterSecret', 'UnlockLightRequest')
             OR lcpm1.signed_message IS NULL)
```
(not p or q is equivalent to p -> q, preventing us from the inner select I think)

Reply:
It should be

```
AND (NOT lcpm1.message_type IN ('SettlementRequired', 'RequestRegisterSecret', 'UnlockLightRequest')
             AND lcpm1.signed_message IS NOT NULL)
```

right? 

Ignore the settlement required, register secret and unlock light when signed message is already present

Reply:
Your proposal isn't quiet right @mauricioirace with your SQL you are saying that you want all messages that have not message type in ('SettlementRequired', 'RequestRegisterSecret', 'UnlockLightRequest') or signed_message null, so if you have a message with type SettlementRequired or you have a message with signed_message not null this query will not retrieve that, basically you need to query a subset and retrieve all the records that are not in that set for the particular case (not always), make sense? that's why i used the nested query, also this is SQLite we don't have all the capabilities from SQL to make the query the best way possible, a better approach to this could be using UNION for example but doesn't work well with SQLite.

Reply:
What about
```

AND (NOT lcpm1.message_type IN ('SettlementRequired', 'RequestRegisterSecret', 'UnlockLightRequest')
             ABD lcpm1.signed_message IS NOT NULL)

```
?

Reply:
No, it doesnt work, it doesnt count the payments msgs.

Reply:
If have a message that is SettlementRequired, it will retrieved if it has not been signed (signed_message is null). On the other hand, if I have a message that has been signed, it will be retrieved only if it's not one of that list. 

I think that's correct, isn't it?

Reply:
As we review it you are right @mauricioirace sound good, i changed the sql to that proposal, thanks!